ILIAS 10: Add Contact privacy paper (#7430)
ILIAS-eLearning/ILIAS: components/ILIAS/Contact/PRIVACY.md

# Contact Service Privacy

This documentation does not warrant completeness or correctness. Please report any
missing or wrong information using the [ILIAS issue tracker](https://mantis.ilias.de)
or contribute a fix via [Pull Request](docs/development/contributing.md#pull-request-to-the-repositories).

## General Information

> - The "Contact Service" of ILIAS provides users a channel to contact each other.

Reviewer: `The "Contact Service" of ILIAS provides a channel for users to contact each other.`

> - The Component only handles the contact and shares of information via the contact functionality.

Reviewer: `The component only handles the contact and shares of information via the contact functionality.`

>   There are multiple other ways of context related contacts that are not part of the service, like Course Member-Tutor

Reviewer: `There are multiple other ways of context related contacts in ILIAS that are not part of this component, like course member/tutor relations or test correction processes, etc..

  relations or Test corrections.

> - The global activation of the Service can be checked with the

Reviewer: `The global activation of the service can be checked with the`

  "Administration > Communication > Contact > Activate 'Contacts'" setting.

> - The personal activation of the Service can be toggled with the

Reviewer: `The personal activation of the service can be toggled with the`

  "Profile and Privacy > Visibility > Allow Contact Requests" setting.
- Contacts can be requested within member lists, like in courses and groups.

> - Further the profile can be published to give all users access to request contact via the "Who is online" tool.

Reviewer: `Furthermore, user profiles can be published to give all users access to request contact via the "Who is online" tool.`

And: This does not depend on the publication of user profiles. It only depends on the personal "Visibility" setting "Allow Contact Requests" if a User A is able to initiate a contact request for User B.

Furthermore, it is currently not possible to initiate a request in the "Who is online?" tool. You only can "Accept" or "Reject" open requests there.

Author: This should mean that, after a profile is published, the "Who is online" tool provides a link to the profile and therefore also a way to request a contact on the profile page indirectly through a public access point (beside closed access from profiles through course etc).

I will clarify that.

  - This can be activated in your profile under "Profile and Privacy > Publish Profile > Visible for logged-in Users"
    setting.
  - This setting does not allow or prevent the contact request itself.

> - A contact can be used for 2 actions:

Reviewer: `A contact can be used for two actions:`

  - You can add contacts to your mailing lists
  - You can invite contacts to chatrooms

> - Further approved contacts are shown inside the "Who is online" Tool

Reviewer: `Furthermore, approved contacts are shown within the "Who is online?" tool`

> - The "Contact Service" is not aware of the data being passed through mails or chats. It does not know anything about

Reviewer: The "Contact" service is not aware of the data being passed through mails or chats. It does not know anything about

  the origin (i.e. course) of the content or whether it contains user related data or not.

## Data being stored

- Each contact request is stored, including the ID of the requesting user and the ID of the requested user, as well as a
  timestamp of the request time. This does not require the consent of both target users.
- When the requested user actively ignores the request, this data entry is added with an "ignore" flag.
- After confirmation, each contact is stored, including the ID of the requesting user and the ID of the requested user as
  well as a timestamp of the request time.
- The storage of requests and approved contacts are separated.

> - Each contact of a mailing list entry is stored including his user_id.

Reviewer: `Each contact added as an entry to a mailing list is stored by using the user_id.`

- Any user relations that originate from the contact is not related or depending on the active contact. There is no
  traceable relation of contacts withing other Services.

## Data being presented

> - Contact data is presented within "Communication > Contacts" and within a separate Section of the "Who is online" Tool.

Reviewer: `Contact data is presented within "Communication > Contacts" and within a separate section of the "Who is online?" tool.`

> - The presentation includes a presentation of the user's profile (restricted depending on the user's profile publishing settings) and a control structure presenting the user's options for contact requests.

Reviewer: `The presentation includes a presentation of the user's profile (restricted depending on the user's profile publishing settings) and control elements presenting the user's options for contact requests ("Accept/Reject").`

Author: Changed the options to their actual wording ("Confirm contact"/"Ignore contact")

- Initially the options are restricted to requesting a contact.
- The requesting user can cancel the request.
- The requested user has the option to actively ignore or confirm the request.
- Both contacts can unlink the contact without any further consent.

> - Each party can detect a change of the contact relation within its own contact options. The only exception to this

Reviewer: We should use another word for `ignorance`.

Author: Changed to "ignoring" but i would still set to the word root since this is also the translation of the UI option.

  is the active ignorance of a request. The requesting user can not detect if his request was ignored.

> - A list of all contacts is shown as a selection when a user adds members to a newly created Mailing List.

Reviewer: `A list of all contacts is shown as a selection when a user adds members to a newly created mailing list.`

> - A contact relation cannot be observed within the UI by any user other than the contacts themselves.

Reviewer: `A contact relation cannot be perceived within the user interface by any user other than the contacts themselves.`

## Data being deleted

- If a contact request is canceled, the stored entry is removed without any traceable footprint.

> - If a contact request is ignored the flagged, stored entry will be preserved to prevent another request.

Reviewer: `If a contact request is ignored, the flagged entry in the database will be preserved to prevent another request.`

- If a contact is unlinked by one of the contacts, the stored entry is removed without any traceable footprint.
- Any user relations that originate from the contact are not affected by its deletion. This includes preservation in
  mailing lists and participation within chatrooms.

> - If a user is deleted, his contacts, his contact requests, requestions against him and his mailing lists and his 

Reviewer: We should avoid `his` and `him`.

  mailing list participations are removed without any traceable footprint.

## Data being exported

> - The Contact Service does not provide any export option itself.

Reviewer: `The "Contact" service does not provide any export option itself.`

> - A contact request can trigger a OSD Notification which is displayed to the requested user. The notification contains

Reviewer: `A contact request can trigger a notification, which may be displayed as a "Toast" to the requested user. The notification contains`

  the users presentation name (depending on his visibility settings) and the users intention of the contact request.

>   - The notification is handled by the notification Service, is self persitent therefore it is not depending on the

Reviewer: `The notification is handled by the "Notification" component, is self persitent, and therefore not depending on the`

    causing request.

> - Otherwise the information of contacts is not stored in any export of the related Services like Mail, Chatroom or Objects.

Reviewer: `Otherwise the information of contacts is not stored in any export of the related components like "Mail", "Chatroom" or "Object".`

- Contact data is presented within "Communication > Contacts" and within a separate section of the "Who is online?" tool.
- The presentation includes a presentation of the user's profile (restricted depending on the user's profile publishing
  settings) and control elements presenting the user's options for contact requests ("Confirm Contact"/"Ignore Contact").
- Initially the options are restricted to requesting a contact.
- The requesting user can cancel the request.
- The requested user has the option to actively ignore or confirm the request.
- Both contacts can unlink the contact without any further consent.
- Each party can detect a change of the contact relation within its own contact options. The only exception to this
  is the active ignoring of a request. The requesting user can not detect if his request was ignored.
- A list of all contacts is shown as a selection when a user adds members to a newly created mailing list.
- A contact relation cannot be perceived within the user interface by any user other than the contacts themselves.

## Data being deleted

- If a contact request is canceled, the stored entry is removed without any traceable footprint.

> - If a contact request is ignored, the flagged entry in the database will be preserved to prevent another request.

Reviewer: `to prevent another request.` -> maybe better?: `to prevent a repeated contact request from the same initiator`

- If a contact is unlinked by one of the contacts, the stored entry is removed without any traceable footprint.
- Any user relations that originate from the contact are not affected by its deletion. This includes preservation in
  mailing lists and participation within chatrooms.
- If a user is deleted, its contacts, send and received contact requests, mailing lists and mailing list participations
  are removed without any traceable footprint.

## Data being exported

- The component does not provide any export option itself.
- A contact request can trigger a notification, which may be displayed as a "Toast" to the requested user. The 
  notification contains the users presentation name (depending on his visibility settings) and the users intention of
  the contact request.
  - The notification is handled by the "Notification" component, is self persistent, and therefore not depending on the
    origin request.

> - Otherwise, the information of contacts is not stored in any export of the related components like "Mail", "Chatroom" or "ILIASObject".

Reviewer: We should not use `ILIASObject` here. If you mean `Services/Object` user `Object` as component name.

Author: The component `Object` was renamed to `ILIASObject` in ILIAS 10

  traceable relation of contacts withing other components.

## Data being presented

- Contact data is presented within "Communication > Contacts" and within a separate section of the "Who is online?" tool.
- The presentation includes a presentation of the user's profile (restricted depending on the user's profile publishing
  settings) and control elements presenting the user's options for contact requests ("Confirm Contact"/"Ignore Contact").
- Initially the options are restricted to requesting a contact.
- The requesting user can cancel the request.
- The requested user has the option to actively ignore or confirm the request.
- Both contacts can unlink the contact without any further consent.
- Each party can detect a change of the contact relation within its own contact options. The only exception to this
  is the active ignoring of a request. The requesting user can not detect if his request was ignored.
- A list of all contacts is shown as a selection when a user adds members to a newly created mailing list.
- The "Look up Users" functionality within the composition of a mail can provide a list of contacts if such matches with
  the search term exist.

> - Furthermore, the "List members" functionality withing the listing of "My Courses" withing the composition of a mail is

Reviewer: ```suggestion
- Furthermore, a list of user accounts with membership assignment is shown by clicking the
 "List Members" action for one or multiple objects within the "My Courses/My Groups"
```

>   provided by the component and shows the contact status to every member.

Reviewer: ```suggestion
  view, accessible directly from the "Compose" view of "Mail" component, or by clicking the "Contacts"
  link item in the main bar. For each account, the username is presented. Additionally, full names are
  presented according to the "Profile and Privacy > Pubslish Profile > Visible for logged in Users"
  setting. The "Contact Status" is shown as well.
```

> - Also, the "Mail to members" functionality in every mail-distributing object provides its user table via the component
>   even if the presented information does not show any information about contact relations.

Reviewer: ```suggestion
- The "Mail to Members" view of membership-enabled object types, which is also located in the "Contact"
 component, provides a list of participating user accounts. For each account,
 the username is presented. Additionally, full names are presented according to
 the "Profile and Privacy > Pubslish Profile > Visible for logged in Users" setting.
 The assigned local roles in the current object context are shown as well.
```

  "Profile and Privacy > Visibility > Allow Contact Requests" setting.
- Contacts can be requested within member lists, like in courses and groups.
- Furthermore, when a user profiles is published, it gives all users access to the profile via the "Who is online" tool
  and, through that, the option to request contact with that profile.
  - This can be activated in your profile under "Profile and Privacy > Publish Profile > Visible for logged-in Users"
    setting.
  - This setting does not allow or prevent the contact request itself.
- A contact can be used for two actions:
  - You can add contacts to your mailing lists
  - You can invite contacts to chatrooms
- Furthermore, approved contacts are shown within the "Who is online?" tool
- The component is not aware of the data being passed through mails or chats. It does not know anything about
  the origin (i.e. course) of the content or whether it contains user related data or not.

## Data being stored

- Each contact request is stored, including the ID of the requesting user and the ID of the requested user, as well as a
  timestamp of the request time. This does not require the consent of both target users.

> - When the requested user actively ignores the request, this data entry is added with an "ignore" flag.

Reviewer 2: I would like to specify "actively ignore". Imo "ignore" suggests that no action is necessary, so "actively ignore" sounds like  "actively doing smth that doesn't request an action". Suggestion: When the requested user actively ignores the request (by choosing the option "Ignore contact")...

  well as a timestamp of the request time.
- The storage of requests and approved contacts are separated.
- Each contact added as an entry to a mailing list is stored by using the user_id.
- Any user relations that originate from the contact is not related or depending on the active contact. There is no
  traceable relation of contacts withing other components.

## Data being presented

- Contact data is presented within "Communication > Contacts" and within a separate section of the "Who is online?" tool.
- The presentation includes a presentation of the user's profile (restricted depending on the user's profile publishing
  settings) and control elements presenting the user's options for contact requests ("Confirm Contact"/"Ignore Contact").
- Initially the options are restricted to requesting a contact.
- The requesting user can cancel the request.
- The requested user has the option to actively ignore or confirm the request.
- Both contacts can unlink the contact without any further consent.
- Each party can detect a change of the contact relation within its own contact options. The only exception to this

>   is the active ignoring of a request. The requesting user can not detect if his request was ignored.

Reviewer 2: Lines 48-52: does this information belong there or more likely in General information? There is no information about presented data  in these lines.

Author: This line is not meant to target the functionality, but the active visible options in the dropdown which are a part of the presentation IMO since they imply certain states of requests.

I tried to clarify this with a more detailed writing. Please let me know what you think of it.

Reviewer 2: Thank you for the detailed description, @iszmais ! 
Just for me to clarify: If an user is responding to a request (by choosing one of these options), then a corresponding data entry will be stored. That's why these options are important, did I understand it correctly? There isn't any personal data visible, but choosing the option triggers the personal data processing, doesn't it?

Author: @ksgribchenko exactly!
And further, the possible options imply the stored data.
E.g. If the option "Unlink" is visible, this implies that a contact relation to this user exists.
Therefore, it provides the same information as displaying the actual request state directly.

  name, if published.
  - This information does not show any information about contact relations.
- Contacts are referenced within the mail service in the sub tab "Contacts".
- A contact relation cannot be perceived within the user interface by any user other than the contacts themselves.

## Data being deleted

- If a contact request is canceled, the stored entry is removed without any traceable footprint.
- If a contact request is ignored, the flagged entry in the database will be preserved to prevent a repeated contact
  request from the same initiator.
- If a contact is unlinked by one of the contacts, the stored entry is removed without any traceable footprint.
- Any user relations that originate from the contact are not affected by its deletion. This includes preservation in
  mailing lists and participation within chatrooms.
- If a user is deleted, its contacts, send and received contact requests, mailing lists and mailing list participations
  are removed without any traceable footprint.

## Data being exported

- The component does not provide any export option itself.

> - A contact request can trigger a notification, which may be displayed as a "Toast" to the requested user. The 

Reviewer 2: I have no problem with Toast, but I have found that less experienced users get confused by it. I think it might be useful to add a short description or rephrase a sentence, maybe smth like: 
"A contact request can trigger a notification, which may be displayed as a "Toast" to the requested user. A "Toast" is a notification element that contains the user presentation name (depending on his visibility) and the user intention of the contact request." 
or (more detailed): 
"A contact request can trigger a notification, which may be displayed as a "Toast" to the requested user. A "Toast" is a notification element that is displayed in the top right-hand corner of the screen and contains the user presentation name (depending on his visibility) and the user intention of the contact request."  - But I'm perfectly fine with "Toast is a notification element".

Author: I choose the short one because I tried to balance the description here. I agree that this should be more explanatory, but at the same time this part should not set properties of the toast and therefore imply a restriction on it. (e.g. "in the top right-hand corner of the screen" is a toast property and should be defined by toast and not the contact, especially since this is not relevant for the contact-concerning part here)

  notification contains the users presentation name (depending on his visibility settings) and the users intention of
  the contact request.

>   - The notification is handled by the "Notification" component, is self persistent, and therefore not depending on the

Reviewer 2: Is it a kind of integrated service? If so, please add a section "Integrated services"/"Services being used" (here are some examples: https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/components/ILIAS/ItemGroup/PRIVACY.md or https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/components/ILIAS/File/PRIVACY.md ) and put this information in this new section. 
If it's not, please ignore this comment.
Some explanation: When I asked you to write privacy.md, the section "Integrated Services" was missing from the"How to write a privacy.md" guide, or did not exist as a separate section. We are in the process of revising the guide and adding this section. We apologize for any inconvenience caused :)

Author: This is more a "Used by services" than "Services being used".

At the end, the notification is a completely independent persistent layer on which the contact has no effect retroactively.

It is like the invitation mail to the course service.

Reviewer 2: okay, got it. Thank you for explaining!

# Contact Service Privacy

This documentation does not warrant completeness or correctness. Please report any
missing or wrong information using the [ILIAS issue tracker](https://mantis.ilias.de)
or contribute a fix via [Pull Request](docs/development/contributing.md#pull-request-to-the-repositories).

## General Information

- The "Contact" component of ILIAS provides a channel for users to contact each other.
- The component only handles the contact and shares of information via the contact functionality.
  There are multiple other ways of context related contacts in ILIAS that are not part of this component, like course 

>   member/tutor relations or test correction processes, etc

Reviewer 3: When I read your version, I had problems understanding the part "shares of information."

Can you slightly change to: 
"The component only handles the contact and sharing of information via the contact functionality. There are multiple other types of context-related contacts in ILIAS that are not part of this component, such as course member/tutor relations or test correction processes, etc."

- The global activation of the component can be checked with the
  "Administration > Communication > Contact > Activate 'Contacts'" setting.
- The personal activation of the component can be toggled with the
  "Profile and Privacy > Visibility > Allow Contact Requests" setting.
- Contacts can be requested within member lists, like in courses and groups.

> - Furthermore, when a user profiles is published, it gives all users access to the profile via the "Who is online" tool

Reviewer 3: Please correct: "a user profiles" to "a user profile" or "user profiles".

  and, through that, the option to request contact with that profile.
  - This can be activated in your profile under "Profile and Privacy > Publish Profile > Visible for logged-in Users"
    setting.
  - This setting does not allow or prevent the contact request itself.
- A contact can be used for two actions:
  - You can add contacts to your mailing lists
  - You can invite contacts to chatrooms

> - Furthermore, approved contacts are shown within the "Who is online?" tool

Reviewer 3: Please add "." at the end of the sentence.

> - The component is not aware of the data being passed through mails or chats. It does not know anything about

Reviewer 3: Can you slightly change to:
"The component does not track or recognize the data being exchanged through emails or chats. It has no information about the source of the content (i.e. a course) or whether it includes user-related data."

  the origin (i.e. course) of the content or whether it contains user related data or not.

## Data being stored

- Each contact request is stored, including the ID of the requesting user and the ID of the requested user, as well as a
  timestamp of the request time. This does not require the consent of both target users.
- When the requested user actively ignores the request (by choosing the option "Ignore contact"), this data entry is
  added with an "ignore" flag.
- After confirmation, each contact is stored, including the ID of the requesting user and the ID of the requested user as
  well as a timestamp of the request time.
- The storage of requests and approved contacts are separated.

> - Each contact added as an entry to a mailing list is stored by using the user_id.

Reviewer 3: You can delete "by" here.

> - Any user relations that originate from the contact is not related or depending on the active contact. There is no

Reviewer 3: Can you slightly change to:
"Any user relations that originate from the contact are not related to or dependent on the active contact. There is no traceable relation of contacts within other components."

  traceable relation of contacts withing other components.

## Data being presented

> - Contact data is presented within "Communication > Contacts" and within a separate section of the "Who is online?" tool.

Reviewer 3: You can change "within" to "in".

> - The presentation includes a presentation of the user's profile (restricted depending on the user's profile publishing

Reviewer 2: ```suggestion
- The presentation includes a presentation of the user's profile (at least the username, other information are restricted depending on the user's profile publishing settings)
```
afaik the user name is always displayed, even if a requesting user does not make their profile public.

  Users" setting. The "Contact Status" is shown as well.
- The "Mail to Members" view of membership-enabled object types, which is also located in the "Contact" component,
  provides a list of participating user accounts. For each account, the username is presented and additionally, the full
  name, if published.
  - This information does not show any information about contact relations.
- Contacts are referenced within the mail service in the sub tab "Contacts".
- A contact relation cannot be perceived within the user interface by any user other than the contacts themselves.

## Data being deleted

- If a contact request is canceled, the stored entry is removed without any traceable footprint.
- If a contact request is ignored, the flagged entry in the database will be preserved to prevent a repeated contact
  request from the same initiator.
- If a contact is unlinked by one of the contacts, the stored entry is removed without any traceable footprint.
- Any user relations that originate from the contact are not affected by its deletion. This includes preservation in
  mailing lists and participation within chatrooms.

> - If a user is deleted, its contacts, send and received contact requests, mailing lists and mailing list participations

Reviewer 3: Can you slightly change to:
"If a user is deleted, their contacts, sent and received contact requests, mailing lists, and mailing list participations are removed without any traceable footprint."

  are removed without any traceable footprint.

## Data being exported

- The component does not provide any export option itself.

> - A contact request can trigger a notification, which may be displayed as a "Toast" to the requested user. A "Toast" is

Reviewer 3: Can you slightly change to:
A contact request can trigger a notification, which may be displayed as a "Toast" to the requested user. A "Toast" is a notification element that contains the user's presentation name (depending on their visibility settings) and the user's intention for the contact request.
The notification is handled by the "Notification" component, is self-persistent, and therefore not dependent on the origin request.
